kano_updater/commands/install.py

```diff
@@ -225,7 +225,10 @@ def install_pip_packages(progress):
 
         success = run_pip_command("install --upgrade {}".format(pkg))
 
-        # TODO: abort the install?
         if not success:
             msg = "Installing the '{}' pip package failed".format(pkg)
             logger.error(msg)
+            if not is_internet():
+                msg="Network is down, aborting PIP install"
+                logger.error(msg)
+                raise IOError(msg)
```

Comment:
Good point @tombettany . It is possible to tell pip to install in 2 stages `pip install --download /var/somewhere/pips` followed by `pip install --no-index --find-links=file:///var/somewhere/pips`. Skipping for now, but sharing this for later implementation.